react/react1/week3/week3homework/src/Components/TodoList.js

```diff
@@ -0,0 +1,108 @@
+import { useState, useEffect } from "react";
+import FancyBorder from "./FancyBorder";
+
+function TodoList({ todos, onDelete, onUpdate }) {
+  const [startEditable, setStartEditable] = useState(false);
+  //   const [endEditable, setEndEditable] = useState(false);
+  const [inputValue, setInputValue] = useState(" ");
+
+  const startEditableTodoHandler = () => {
+    setStartEditable(true);
+  };
+  const endEditableTodoHandler = (todoId, todoDeadline, todoDesc) => {
+    // if the description is updated with making any changes it changes to empty string so this is for that case
+    const isDescriptionEmpty = {
+      description: inputValue.length === 0 ? todoDesc : inputValue,
+    };
+    const todo = [
+      {
+        id: todoId,
+        description:
+          isDescriptionEmpty.description.trim().length === 0
+            ? todoDesc
+            : isDescriptionEmpty.description,
+        deadline: todoDeadline,
+      },
+    ];
+    onUpdate(todo);
+    setStartEditable(false);
+  };
+  const updateChangeHandler = (e) => {
+    setInputValue(e.target.value);
+  };
+  if (!startEditable) {
+    return (
+      <div>
+        {todos.map((todo) => {
+          return (
+            <FancyBorder key={todo.id}>
+              <input type="checkbox" className="checkbox" />
+              <p>
+                Description: {todo.description} | Deadline: {todo.deadline}
+              </p>
+
+              <button
+                type="button"
+                onClick={() => onDelete(todo.id)}
+                className="delete"
+              >
+                Delete
+              </button>
+              <button
+                type="button"
+                onClick={startEditableTodoHandler}
+                className="edit"
+              >
+                Edit
+              </button>
+            </FancyBorder>
+          );
+        })}
+      </div>
+    );
+  }
```

Comment:
Try and see if you can somehow avoid the if statement here since the code is duplicated